src/test/java/BaseTest.java

```diff
@@ -1,11 +1,146 @@
+        buttonLogin.click();
+    }
+
+    public void inputSearch(String seachTxt){
+        WebElement seachField = this.driver.findElement(By.cssSelector("input[type='search']"));
+        seachField.sendKeys(seachTxt);
+    }
+
+    public void btnSongsViewAll(){
+        WebElement buttonViewAll = this.driver.findElement(By.xpath("//section[@class='songs']//button[contains(text(),'View All')]"));
+        buttonViewAll.click();
+    }
+
+    public void btnAddTo(){
+        WebElement buttonAddTo = this.driver.findElement(By.cssSelector("button[class='btn-add-to']"));
+        buttonAddTo.click();
+    }
+
+    public void btnPlaySong(){
+        WebElement buttonPlaySong = this.driver.findElement(By.cssSelector("span[data-testid='play-btn']"));
+        buttonPlaySong.click();
+    }
+
+    public void btnPlayNextSong(){
+        WebElement buttonPlayNext = this.driver.findElement(By.cssSelector("i[data-testid='play-next-btn']"));
+        buttonPlayNext.click();
+    }
+
+    public void imgSoundBarVisible(){
+        WebElement buttonPlayNext = this.driver.findElement(By.cssSelector("div[data-testid='sound-bar-play']"));
+        buttonPlayNext.isDisplayed();
+    }
+
+    public void rightClickPlaylistMenu(String playlist){
+        WebElement delPlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id=\"playlists\"]//a[contains(text(),\"" + playlist + "\")]")));
+        action.contextClick(delPlaylist).perform(); // Right click
+    }
+
+    public void selectSubMenuPlaylistOption(String option){
+        WebElement liDeleteTxt = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//nav//li[contains(text(),\""+ option +"\")]")));
+        liDeleteTxt.click();
+    }
+
+    public void updatePlaylistTextBoxContent(String playlistName) {
+        WebElement txtEditPlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@data-testid=\"inline-playlist-name-input\"]")));
+        //txtEditPlaylist.clear(); It doesnt work - when I use clear right after the txtbox will become hidden
+        txtEditPlaylist.sendKeys(Keys.chord(Keys.CONTROL,"a"),playlistName);
+        txtEditPlaylist.sendKeys(Keys.ENTER);
+    }
+
+    // creates playlist
+    public void createPlaylist(String playlistName)  {
+
+        WebElement btnPlusIcon = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("i[data-testid=\"sidebar-create-playlist-btn\"]")));
+        btnPlusIcon.click();
+
+        WebElement liAddNewPlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li[data-testid=\"playlist-context-menu-create-simple\"")));
+        liAddNewPlaylist.click();
+
+        WebElement txtPlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@name=\"name\"]")));
+        txtPlaylist.sendKeys(playlistName);
+        txtPlaylist.sendKeys(Keys.ENTER);
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.

For example
![Screenshot_2](https://github.com/YuriiLyndiuk/A57-QA-Automation-Framework/assets/104761482/b07cd7c1-7977-4e53-80ea-8045ac17f779)